KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/LiveStreamSettingsModule.java

```diff
@@ -0,0 +1,308 @@
+package com.kaltura.media_server.modules;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.kaltura.client.types.KalturaLiveEntry;
+import com.kaltura.media_server.services.Utils;
+import com.wowza.wms.amf.*;
+import com.wowza.wms.application.*;
+import com.wowza.wms.httpstreamer.cupertinostreaming.livestreampacketizer.*;
+import com.wowza.wms.media.mp3.model.idtags.*;
+import com.wowza.wms.stream.livepacketizer.*;
+import com.wowza.wms.stream.IMediaStream;
+import com.wowza.wms.module.*;
+import com.wowza.wms.stream.*;
+import com.wowza.wms.client.IClient;
+import com.wowza.wms.application.WMSProperties;
+import com.wowza.wms.stream.MediaStreamActionNotifyBase;
+import org.apache.log4j.Logger;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.text.SimpleDateFormat;
+import java.lang.Thread;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+// todo: test and add relevant code ofr ptsTimeCode moving back in time and PTS wrap arround are supported.
+
+public class LiveStreamSettingsModule extends ModuleBase {
+
+	private static final String OBJECT_TYPE_KEY = "objectType";
+	private static final String OBJECT_TYPE_SYNCPOINT = "KalturaSyncPoint";
+	private static final String HEADER_TAG_TIMECODE = "EXT-KALTURA-SYNC-POINT";
+	private static final String TIMESTAMP_KEY = "timestamp";
+	private static final String ID_KEY = "id";
+	private static final Logger logger = Logger.getLogger(LiveStreamSettingsModule.class);
+	private static final String MAX_ALLOWED_PTS_DRIFT_MILLISEC = "KalturaMaxAllowedPTSDriftiMillisec";
+	private static final int DEFAULT_MAX_ALLOWED_PTS_DRIFT_MILLISEC = 1000;
+
+	private LiveStreamPacketizerListener liveStreamPacketizerListener;
+	private int maxAllowedPTSDriftMillisec;
+	private ConcurrentHashMap<String, Long> mapLiveEntryToBaseSystemTime = null;
+	private ConcurrentHashMap<String, Long> mapLiveEntryStreamsCount = null;
+	private ConcurrentHashMap<String, PacketListener> packetListeners = null;
```

Comment:
too complex, it's easier to save it in the stream property, please look at the many wowza examples